### PR TITLE
Adapt to the new validator network

### PR DESCRIPTION
**Pull request by piotrMocz**
Introduces two new arguments: `--ip` and `--dns`, one of which is always required. The `PUBLIC_ADDR` and `PUBLIC_VALIDATOR_ADDRESS` vars will be deduced from the provided IP/DNS and ports.

### CODE REVIEW

File: run_node.sh [hunk 1]
```diff
@@ -111,6 +126,18 @@ then
...
+        if [[ -n "${PUBLIC_DNS}" ]]
+        then
+            PUBLIC_ADDR="/dns4/${PUBLIC_DNS}/tcp/${PORT}"
+            PUBLIC_VALIDATOR_ADDRESS="${PUBLIC_DNS}:${VALIDATOR_PORT}"
+        else
+            PUBLIC_ADDR="/ip4/${PUBLIC_IP}/tcp/${PORT}"
+            PUBLIC_VALIDATOR_ADDRESS="${PUBLIC_IP}:${VALIDATOR_PORT}"
+        fi
+
+        echo "Running with public address: ${PUBLIC_ADDR}"
```
Comment (timorl) [inline run_node.sh:140]: Perhaps print both public addresses here.

**Review by timorl — APPROVED** [review]
Good stuff.

**Review by maciejnems — APPROVED** [review]
Very nice.

File: run_node.sh [hunk 2]
```diff
@@ -5,10 +5,12 @@ set -eo pipefail
 Help()
 {
     echo "Run the aleph-node as either a validator or an archivist."
-    echo "Syntax: ./run_node.sh [--<name|image|stash_account> <value>] [--<archivist|mainnet|build_only|sync_from_genesis>]"
+    echo "Syntax: ./run_node.sh [--<name|image|stash_account|ip|dns> <value>] [--<archivist|mainnet|build_only|sync_from_genesis>]"
     echo
     echo "options:"
     echo "archivist         Run the node as an archivist (the default is to run as a validator)."
+    echo "ip                The public IP of your node."
```
[the remark this answers is not in the document]
Comment (timorl) [reply run_node.sh:12]: The public one technically doesn't have to be IPv4, it only matters for local listening. Adding it here would be unnecessarily confusing.

[the remark this answers is not in the document]
Comment (maciejnems) [reply run_node.sh:12]: True, true :facepalm:

[the remark this answers is not in the document]
Comment (piotrMocz) [reply run_node.sh:12]: But if it were a v6, then the `PUBLIC_ADDR` would have to start with `ip6`, right?

[the remark this answers is not in the document]
Comment (timorl) [reply run_node.sh:12]: Ohh, right, thats... just confusing then. :/ We should probably have a proper task for explicit ipv6 support everywhere, unless you think just adding two flags here and constructing the appropriate libp2p address is good enough for now?

**Pull request merged**